panel/lxqtpanel.cpp

```diff
@@ -374,7 +372,7 @@ int LXQtPanel::getReserveDimension()
     return mHidable ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize);
 }
 
-void LXQtPanel::setPanelGeometry(bool animate)
+void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
```

Comment:
> OK, I did some tests. Removal of that "flash" is possible but, apparently, it doesn't worth the effort ...

I did dig into that also... the flicker is there because of the `QDesktopWidget::workAreaResized' signal and our `realign()' on that... I believe we should not handle that signal at all (we don't care about the available geometry and we are setting it)... and now I remember that @RalfJung already pointed this out in #310.
With this patch:
```diff
diff --git a/panel/lxqtpanel.cpp b/panel/lxqtpanel.cpp
index 90bae87..4f9e94a 100644
--- a/panel/lxqtpanel.cpp
+++ b/panel/lxqtpanel.cpp
@@ -187,7 +187,6 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
     mHideTimer.setInterval(PANEL_HIDE_DELAY);
     connect(&mHideTimer, SIGNAL(timeout()), this, SLOT(hidePanelWork()));
 
-    connect(QApplication::desktop(), SIGNAL(workAreaResized(int)), this, SLOT(realign()));
     connect(QApplication::desktop(), SIGNAL(screenCountChanged(int)), this, SLOT(ensureVisible()));
     connect(LXQt::Settings::globalSettings(), SIGNAL(settingsChanged()), this, SLOT(update()));
     connect(lxqtApp, SIGNAL(themeChanged()), this, SLOT(realign()));
@@ -206,8 +205,8 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
     // show it the first time, despite setting
     if (mHidable)
     {
-      showPanel(true);
-      QTimer::singleShot(PANEL_HIDE_FIRST_TIME, this, SLOT(hidePanel()));
+        showPanel(true);
+        QTimer::singleShot(PANEL_HIDE_FIRST_TIME, this, SLOT(hidePanel()));
     }
 }
 
@@ -377,7 +376,7 @@ int LXQtPanel::getReserveDimension()
     return mHidable ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize);
 }
 
-void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
+void LXQtPanel::setPanelGeometry(bool animate)
 {
     const QRect currentScreen = QApplication::desktop()->screenGeometry(mActualScreenNum);
     QRect rect;
@@ -385,7 +384,7 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
     if (isHorizontal())
     {
         // Horiz panel ***************************
-        rect.setHeight(qMax(PANEL_MINIMUM_SIZE, mPanelSize));
+        rect.setHeight(mHidden ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize));
         if (mLengthInPercents)
             rect.setWidth(currentScreen.width() * mLength / 100.0);
         else
@@ -416,24 +415,14 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
 
         // Vert .......................
         if (mPosition == ILXQtPanel::PositionTop)
-        {
-            if (mHidden && !firstTime)
-                rect.moveBottom(currentScreen.top() + PANEL_HIDE_SIZE);
-            else
-                rect.moveTop(currentScreen.top());
-        }
+            rect.moveTop(currentScreen.top());
         else
-        {
-            if (mHidden && !firstTime)
-                rect.moveTop(currentScreen.bottom() - PANEL_HIDE_SIZE);
-            else
-                rect.moveBottom(currentScreen.bottom());
-        }
+            rect.moveBottom(currentScreen.bottom());
     }
     else
     {
         // Vert panel ***************************
-        rect.setWidth(qMax(PANEL_MINIMUM_SIZE, mPanelSize));
+        rect.setWidth(mHidden ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize));
         if (mLengthInPercents)
             rect.setHeight(currentScreen.height() * mLength / 100.0);
         else
@@ -464,19 +453,9 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
 
         // Horiz ......................
         if (mPosition == ILXQtPanel::PositionLeft)
-        {
-            if (mHidden && !firstTime)
-                rect.moveRight(currentScreen.left() + PANEL_HIDE_SIZE);
-            else
-                rect.moveLeft(currentScreen.left());
-        }
+            rect.moveLeft(currentScreen.left());
         else
-        {
-            if (mHidden && !firstTime)
-                rect.moveLeft(currentScreen.right() - PANEL_HIDE_SIZE);
-            else
diff --git a/panel/lxqtpanel.cpp b/panel/lxqtpanel.cpp
index 90bae87..4f9e94a 100644
--- a/panel/lxqtpanel.cpp
+++ b/panel/lxqtpanel.cpp
@@ -187,7 +187,6 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
     mHideTimer.setInterval(PANEL_HIDE_DELAY);
     connect(&mHideTimer, SIGNAL(timeout()), this, SLOT(hidePanelWork()));
 
-    connect(QApplication::desktop(), SIGNAL(workAreaResized(int)), this, SLOT(realign()));
     connect(QApplication::desktop(), SIGNAL(screenCountChanged(int)), this, SLOT(ensureVisible()));
     connect(LXQt::Settings::globalSettings(), SIGNAL(settingsChanged()), this, SLOT(update()));
     connect(lxqtApp, SIGNAL(themeChanged()), this, SLOT(realign()));
@@ -206,8 +205,8 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
     // show it the first time, despite setting
     if (mHidable)
     {
-      showPanel(true);
-      QTimer::singleShot(PANEL_HIDE_FIRST_TIME, this, SLOT(hidePanel()));
+        showPanel(true);
+        QTimer::singleShot(PANEL_HIDE_FIRST_TIME, this, SLOT(hidePanel()));
     }
 }
 
@@ -377,7 +376,7 @@ int LXQtPanel::getReserveDimension()
     return mHidable ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize);
 }
 
-void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
+void LXQtPanel::setPanelGeometry(bool animate)
 {
     const QRect currentScreen = QApplication::desktop()->screenGeometry(mActualScreenNum);
     QRect rect;
@@ -385,7 +384,7 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
     if (isHorizontal())
     {
         // Horiz panel ***************************
-        rect.setHeight(qMax(PANEL_MINIMUM_SIZE, mPanelSize));
+        rect.setHeight(mHidden ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize));
         if (mLengthInPercents)
             rect.setWidth(currentScreen.width() * mLength / 100.0);
         else
@@ -416,24 +415,14 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
 
         // Vert .......................
         if (mPosition == ILXQtPanel::PositionTop)
-        {
-            if (mHidden && !firstTime)
-                rect.moveBottom(currentScreen.top() + PANEL_HIDE_SIZE);
-            else
-                rect.moveTop(currentScreen.top());
-        }
+            rect.moveTop(currentScreen.top());
         else
-        {
-            if (mHidden && !firstTime)
-                rect.moveTop(currentScreen.bottom() - PANEL_HIDE_SIZE);
-            else
-                rect.moveBottom(currentScreen.bottom());
-        }
+            rect.moveBottom(currentScreen.bottom());
     }
     else
     {
         // Vert panel ***************************
-        rect.setWidth(qMax(PANEL_MINIMUM_SIZE, mPanelSize));
+        rect.setWidth(mHidden ? PANEL_HIDE_SIZE : qMax(PANEL_MINIMUM_SIZE, mPanelSize));
         if (mLengthInPercents)
             rect.setHeight(currentScreen.height() * mLength / 100.0);
         else
@@ -464,19 +453,9 @@ void LXQtPanel::setPanelGeometry(bool animate, bool firstTime)
```
everything (also the initilal show-up animation) is working as a charm